src/ThirdPartySupport/Elementor/Widgets/V2/ElementorDonationFormWidget/ElementorDonationFormWidget.php

```diff
@@ -0,0 +1,211 @@
+            'options' => [],
+            'default' => !empty($formOptionsGroup) ? array_key_first($formOptionsGroup[0]['options']) : '',
+            'groups' => $formOptionsGroup,
+        ]);
+
+        $this->add_control(
+            'display_style',
+            [
+                'label' => __('Display Style', 'give'),
+                'type' => \Elementor\Controls_Manager::SELECT,
+                'options' => ['onpage' => __('On Page', 'give'), 'modal' => __('Modal', 'give'), 'newTab' => __('New Tab', 'give')],
+                'default' => 'onpage',
+                'frontend_available' => true,
+            ]
+        );
+
+        $this->add_control(
+            'donate_button_text',
+            [
+                'label' => __('Donate Button Text', 'give'),
+                'type' => \Elementor\Controls_Manager::TEXT,
+                'default' => __('Continue to Donate', 'give'),
+                'frontend_available' => true,
+                'condition' => [
+                    'display_style!' => 'onpage',
+                ],
+            ]
+        );
+
+        $this->end_controls_section();
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function getFormOptionsWithCampaigns(): array
+    {
+        $campaignsWithForms = $this->getCampaignsWithForms();
+
+        if (empty($campaignsWithForms)) {
+            return [];
+        }
+
+        $campaignOptions = [];
+        $formOptionsGroup = [];
+        $campaignGroups = [];
+
+        // Group forms by campaign
+        foreach ($campaignsWithForms as $item) {
+            // Skip items without campaign association
+            if (empty($item->campaign_id) || empty($item->campaign_title)) {
+                continue;
+            }
+
+            $campaignId = $item->campaign_id;
+            $campaignTitle = $item->campaign_title;
+
+            // Add to campaign options if not already added
+            if (!isset($campaignOptions[$campaignId])) {
+                $campaignOptions[$campaignId] = $campaignTitle;
+                $campaignGroups[$campaignId] = [
+                    'label' => $campaignTitle,
+                    'options' => []
+                ];
+            }
+
+            // Add form to the campaign group
+            $campaignGroups[$campaignId]['options'][$item->id] = $item->title;
+        }
+
+        $formOptionsGroup = array_values($campaignGroups);
+
+        return $formOptionsGroup;
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function render(): void
+    {
+        $settings = $this->get_settings_for_display();
+        $displayStyle = $settings['display_style'];
+        $donateButtonText = $settings['donate_button_text'];
+        $formId = $settings['form_id'];
+
+        if (empty($formId)) {
+            return;
+        }
+
+        echo do_shortcode(sprintf('[give_form display_style="%s" continue_button_title="%s" id="%s"]', $displayStyle, $donateButtonText, $formId));
```

Comment:
Should we use `esc_attr` for it? 

Example: 
```
echo do_shortcode(sprintf('[give_form display_style="%s" continue_button_title="%s" id="%s"]', esc_attr($displayStyle), esc_attr($donateButtonText), esc_attr($formId)));
```

Reply:
Yup good catch, although I think these are being escaped before output within the shortcode logic but doesn't hurt to add here as well

Resolved 3d408e36